documentation/library_guide/parallel_api/pass_data_algorithms.rst

```diff
@@ -81,7 +81,16 @@ the buffer were created for the same queue. For example:
     return 0;
   }
 
-Alternatively, use ``std::vector`` with a USM allocator. For example:
+Alternatively, use ``std::vector`` with a USM allocator.
+
+Note: The ability to appropriately detect USM allocated ``std::vector::iterator`` depends
+on details of the C++ standard library implementation and what information about the
+allocator is included in the ``std::vector::iterator`` type definition. If USM allocated
+vector iterators are not detectable with your C++ standard library, they will still function
+as inputs to oneDPL, but they will be treated as if they were host-side 
```

Comment:
Probably, it makes sense to add the clarification that in case "USM allocated
vector iterators are not detectable with your C++ standard library" a user can use `vec.data()` instead of "vec.begin()". It guarantees data processing w/o host-device extra copying.

Reply:
Yes, this is a good idea.

Reply:
After offline discussion, I've made these changes, but also separated out the documentation changes into another PR #1477.